core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -623,22 +617,34 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>,
                                         : a_functor) num_teams(nteams) \
     thread_limit(team_size)
     for (idx_type team_id = 0; team_id < n_chunks; ++team_id) {
+      typename Analysis::Reducer final_reducer(&a_functor);
 #pragma omp parallel num_threads(team_size)
       {
         const idx_type local_offset = team_id * chunk_size;
         value_type offset_value;
         if (team_id > 0)
           offset_value = chunk_values(team_id - 1);
         else
-          ValueInit::init(a_functor, &offset_value);
+          final_reducer.init(&offset_value);
 
 #pragma omp for
         for (idx_type i = 0; i < chunk_size; ++i) {
           const idx_type idx = local_offset + i;
           value_type local_offset_value;
           if (i > 0) {
             local_offset_value = element_values(team_id, i - 1);
-            ValueJoin::join(a_functor, &local_offset_value, &offset_value);
+            // FIXME We seem to access memory illegaly on AMD GPUs
```

Comment:
Just FYI - we are following the convention of `FIXME_OPENMPTARGET` if the issue is something related to OpenMPTarget.